### PR TITLE
ci(lint): bump golangci-lint to 1.60.3

### PR DESCRIPTION
**Pull request by mikeee**
# Description

This bumps the golangci-lint to v1.60.3, bringing a few new linters and getting us to speed with deprecations/changes.

Issues for new linters and their associated errors, they will be disabled in this PR as it is more succinct and allows for discussions about changes required not purely related to the linter implementation.
- #8025
- #8026 
- #8027 
- #8028 
- #8029 
- #8033 
- #8034 
- #8035 
- #8036 

Changes to non-linter code:

Change details are available in the extended commit messages.

## Issue reference

Please reference the issue this PR will close: #_N/A_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by mikeee** [comment]
This PR will be made ready for review once I address the issues raised by the updated linters.

File: pkg/channel/http/http_channel_test.go [hunk 1]
```diff
@@ -68,7 +69,14 @@ func (t *testHandlerHeaders) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 	rsp, _ := json.Marshal(headers)
-	io.WriteString(w, string(rsp))
+
+	var rspBuffer bytes.Buffer
+	if _, err := rspBuffer.Write(rsp); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	io.WriteString(w, rspBuffer.String())
```
Comment (mukundansundar) [inline pkg/channel/http/http_channel_test.go:79]: was this error check being highlighted with the newer linter?

Comment (mikeee) [reply pkg/channel/http/http_channel_test.go:79]: It was indeed as a staticcheck (SA6006 - Using io.WriteString to write []byte), I just clocked that I can just use the response writer - sleepy moment 😅

**Review by mukundansundar — APPROVED** [review]

**Comment by mikeee** [comment]
@dapr/maintainers-dapr - Any appetite to get this merged or should I close as the list of conflicts is growing

**Comment by mikeee** [comment]
Closing in favour of #8146

**Pull request closed without merging**